templates/home/home.html.twig

```diff
@@ -102,69 +103,115 @@
+                        }
+
+                        let track = new AppTrack(map, polylineCollection);
+                        track.setType(tracksData[i].type);
 
-                    var polylineCollection = [];
-                    for (let j = 0; j < tracksData[i].points.length; j++) {
-                        let points = tracksData[i].points[j];
-                        let pointsAsPolyline = [];
-                        for (let pointIndex = 0; pointIndex < points.length; pointIndex++) {
-                            pointsAsPolyline.push([
-                                points[pointIndex].lat,
-                                points[pointIndex].lng
-                            ]);
+                        switch (track.getType()) {
+                            case 1:
+                                if (mapOptions.showByBike) {
+                                    track.show();
+                                }
+                                break;
+                            case 2:
+                                if (mapOptions.showByFoot) {
+                                    track.show();
+                                }
+                                break;
+                            default:
+                                track.show();
                         }
 
-                        let polylineColor = gpsColors[Math.floor(Math.random()*gpsColors.length)];
-                        let polyline = L.polyline(pointsAsPolyline, {color: polylineColor});
-                        polylineCollection.push(polyline);
-                        polyline.bindPopup('<a href="/{{ app.request.locale }}/gps/view/' + encodeURIComponent(slugOrId) + '">{{ 'View details'|trans|e('js') }}</a>');
+                        loadedTracks[id] = track;
                     }
 
-                    let track = new AppTrack(map, polylineCollection);
-                    track.setType(tracksData[i].type);
+                    if (responseStatus == 2) {
+                        moveEndListener();
+                    }
+                })(responseAsJson.tracks);
 
-                    switch (track.getType()) {
-                        case 1:
-                            if (mapOptions.showByBike) {
-                                track.show();
-                            }
-                            break;
-                        case 2:
-                            if (mapOptions.showByFoot) {
-                                track.show();
-                            }
-                            break;
-                        default:
-                            track.show();
+                // places
+                (function (data) {
+                    const responseStatus = data.status;
+
+                    if (responseStatus != 1 && responseStatus != 2) {
```

Comment:
These are also not friendly.

Reply:
I don't know what is the right way to handle this in JS